packages/assets-controllers/src/MultichainAssetsRatesController/MultichainAssetsRatesController.ts

```diff
@@ -315,6 +324,34 @@ export class MultichainAssetsRatesController extends StaticIntervalPollingContro
     });
   }
 
+  /**
+   * Fetches historical prices for the current account
+   *
+   * @param asset - The asset to fetch historical prices for.
+   * @returns The historical prices.
+   */
+  async fetchHistoricalPrices(
+    asset: CaipAssetType,
+  ): Promise<OnAssetHistoricalPriceResponse> {
+    const selectedAccount = this.messagingSystem.call(
+      'AccountsController:getSelectedMultichainAccount',
+    );
+
+    const currentCurrency =
+      MAP_CAIP_CURRENCIES[this.#currentCurrency] ?? MAP_CAIP_CURRENCIES.usd;
+
+    const historicalPrices = await this.#handleSnapRequest({
+      snapId: selectedAccount?.metadata.snap?.id as SnapId,
+      handler: HandlerType.OnAssetHistoricalPrice,
+      params: {
+        from: asset,
+        to: currentCurrency,
+      },
+    });
+
+    return historicalPrices as OnAssetHistoricalPriceResponse;
+  }
+
```

Comment:
Don't think it is necessary to save these prices in state. Calling this function can be driven only by the user clicking on the asset. Prices can instead be cached on UI side and we make sure this function is only called when the prices time has expired.

Pseudo code

```
const currentTime = Date.now();
const { cachedHistoricalPrices } = (await getStorageItem(cacheKey)) || {}; 
// the cacheKey can be a combination between the asset and the currency we fetched prices for exp : asset:currency
if (cachedResponse && (currentTime < cachedHistoricalPrices?.expirationTime) {
    return cachedResponse;
 }
// call fetchHistoricalPrices from multichainAssetsRatesController and update the data for the cacheKey.
```

Reply:
Wouldn't it be better to use the controller state but don't persist this key that way we can defer that logic of caching and validating the cached value here ? In my opinion it would be more elegant.

Reply:
The controller state doesn't need to be necessarily persisted, we can utilize it as a cache to store the historical prices under a new property like `historicalPrices: Record<CaipAssetType, HistoricalPrice>` and make this state property not persisted int he state that way it gets cleared when the client reboots

Reply:
Do we have other controllers already implementing cache without persisting state, maybe this https://github.com/MetaMask/core/blob/a6937e35b4ddef7c307e5b03877e2d54bfc3e3c9/packages/profile-sync-controller/src/shared/encryption/cache.ts#L32 ?